napari/util/keybindings.py

```diff
@@ -0,0 +1,582 @@
+                keys.BACKSPACE,
+                keys.F1,
+                keys.F2,
+                keys.F3,
+                keys.F4,
+                keys.F5,
+                keys.F6,
+                keys.F7,
+                keys.F8,
+                keys.F9,
+                keys.F10,
+                keys.F11,
+                keys.F12,
+                keys.SPACE,
+                keys.ENTER,
+                keys.TAB]
+
+MODIFIER_KEYS = OrderedDict(C=keys.CONTROL,
+                            M=keys.ALT,
+                            S=keys.SHIFT,
+                            s=keys.META)
+
+
+shorthand_modifier_patt = re.compile(f"([{''.join(MODIFIER_KEYS)}])(?=-)")
+
+
+class BindOperation(Enum):
+    BIND = 'bind'
+    UNBIND = 'unbind'
+    REBIND = 'rebind'
+
+
+def _expand_shorthand(shorthand_seq):
+    """Expand shorthand form of a key sequence.
+
+    C -> Control
+    M -> Alt
+    S -> Shift
+    s -> Meta
+    """
+    return re.sub(shorthand_modifier_patt,
+                  lambda m: MODIFIER_KEYS[m.group(0)].name,
+                  shorthand_seq)
+
+
+def parse_seq(seq):
+    """Parse a key sequence into its components in a comparable format.
+
+    Parameters
+    ----------
+    seq : str
+        Key sequence.
+
+    Returns
+    -------
+    key : str
+        Base key of the sequence.
+    modifiers : set of str
+        Modifier keys of the sequence.
+    """
+    parsed = re.split('-(?=.+)', seq)
+    modifiers, key = parsed[:-1], parsed[-1]
+
+    return key, set(modifiers)
+
+
+def components_to_seq(key, modifiers):
+    """Combine components to become a key sequence.
+
+    Modifier keys will always be combined in the same order:
+    Control, Alt, Shift, Meta
+
+    Shift pressed with any letter will be consumed
+    to transform it to upper case, e.g. Shift + w => W.
+    Otherwise, the letter will be lower case.
+    Shift will also be consumed without transformation when pressed
+    with any other non-special key unless Control is also a modifier.
+
+    Parameters
+    ----------
+    key : str or vispy.app.Key
+        Base key.
+    modifiers : sequence of str or vispy.app.Key
+        Modifier keys.
+
+    Returns
+    -------
+    seq : str
+        Generated key sequence.
+    """
+    if len(key) == 1 and key.isalpha():  # it's a letter
+        if 'Shift' not in modifiers:
+            key = key.lower()
+            cond = lambda m: True  # noqa: E731
+        else:
+            # Shift is consumed to create upper-case letter
+            key = key.upper()
+            cond = lambda m: m != 'Shift'  # noqa: E731
+    elif key in SPECIAL_KEYS:
+        # remove redundant information i.e. an output of 'Shift-Shift'
+        cond = lambda m: m != key  # noqa: E731
+    else:
+        # Shift is consumed to transform key
+
+        # bug found on OSX: Command will cause Shift to not
+        # transform the key so do not consume it
+        # note: 'Control' is OSX Command key
```

Comment:
This note should probably be in the module-level docstring

Reply:
it is already. did i not make it clear enough?

Reply:
@kne42 I don't know if I'm blind but [here's](https://github.com/napari/napari/pull/251/files#diff-266e2fe0a33dbefd726fb2bf1e11055bR4) the module-level docstring and I don't see anything about Ctrl being macOS Cmd...